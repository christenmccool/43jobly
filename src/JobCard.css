.JobCard {
  margin: 20px 0;
  color: black;
  background-color: white;
  padding: 10px;
  border-radius: 5px;
  border: 1px solid lightgrey;
  box-shadow: -1px 1px 2px lightgrey;
}


.JobCard-title {
  display: inline-block;
  font-size: 1.5em;
  margin-top: 0;
  margin-bottom: 0.5em;
  font-weight: normal;
}

.JobCard-info {
  font-size: 1em;
  margin-top: 0;
  font-weight: normal;  
}