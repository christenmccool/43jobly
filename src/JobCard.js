import React, {useState, useContext} from 'react';
import UserContext from './UserContext';
import {Link} from 'react-router-dom';
import './JobCard.css';

/** JobCard component for Jobly app
 * Renders a card displaying the job title, salary, and equity
 */

const JobCard = ({id, title, salary, equity, companyHandle, companyName, apply}) => {
  const user = useContext(UserContext);
  const initialApplied = user ? user.applications.indexOf(id) !== -1 : false;
  const [applied, setApplied] = useState(initialApplied);

  const handleClick = () => {
    apply(id);
  }

  const appElem = applied ? 
                    <button className="JobCard-btn JobCard-applied" disabled>Applied</button> :
                    <button className="JobCard-btn JobCard-apply-btn" onClick={handleClick}>Apply</button>

  return (
    <div className="JobCard">
      <div>
        <h1 className="JobCard-title">{title}</h1>
        {apply? appElem : null}
      </div>
      {companyName ? <Link exact to={`/companies/${companyHandle}`}><span className="JobCard-subtitle">{companyName}</span></Link> : null}
      <p className="JobCard-info">Salary: {salary}</p>
      <p className="JobCard-info">Equity: {equity}</p>
    </div>
  )
}

export default JobCard;