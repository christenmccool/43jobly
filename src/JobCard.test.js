import { render, asFragment } from '@testing-library/react';
import JobCard from './JobCard';

test('if it renders without crashing', () => {
  render(<JobCard />);
});

test('if it matches snapshot', () => {
  const {asFragment} = render(<JobCard />);
  expect(asFragment()).toMatchSnapshot();
});

test('if it renders title, salary, and equity', () => {
  const {queryByText} = render(<JobCard title="Test title" salary={100} equity={0.5} />);
  expect(queryByText("Test title")).toBeInTheDocument();
  expect(queryByText("Salary: 100")).toBeInTheDocument();
  expect(queryByText("Equity: 0.5")).toBeInTheDocument();
});