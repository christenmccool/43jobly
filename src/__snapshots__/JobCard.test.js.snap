// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`if it matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="JobCard"
  >
    <h1
      class="JobCard-title"
    />
    <p
      class="JobCard-info"
    >
      Salary: 
    </p>
    <p
      class="JobCard-info"
    >
      Equity: 
    </p>
  </div>
</DocumentFragment>
`;
